rest_framework/test.py

```diff
@@ -195,7 +195,7 @@ def _encode_data(self, data, format=None, content_type=None):
                 ret = renderer.render(data)
 
             # Coerce text to bytes if required.
-            if isinstance(ret, str):
+            if isinstance(ret, str) and renderer.charset:
```

Comment:
The test suite passes if we move the existing `if` under the above else:

```python
            if data is None:
                ret = ''
            else:
                ret = renderer.render(data)

                # Coerce text to bytes if required.
                if isinstance(ret, str):
                    ret = ret.encode(renderer.charset)
```

Is it different or equivalent to your fix? Is there a use-case where the regression will remain? If so, would be nice to add isuch case to our test suite...